tesseract/tesseract_visualization/include/tesseract_visualization/trajectory_interpolator.h

```diff
@@ -52,7 +52,7 @@ class TrajectoryInterpolator
 
   tesseract_planning::MoveInstruction getMoveInstruction(double request_duration) const;
 
-  double getMoveInstructionDuration(std::size_t index) const;
```

Comment:
Not upset either way, but duration seems like a size. It seems like you are just trading a cast in trajectory_player for a cast in trajectory interpolator. I wonder if index should just be a size_t. It wouldn't make since to play an index larger than the size it can hold anyway.

Reply:
These are a good read. As a result I plan to avoid using std::size_t in class interface. Let me know what your take is on the information below.

[Google Style Guide](https://google.github.io/styleguide/cppguide.html#Integer_Types)
[Issue Discussion](https://github.com/fish-shell/fish-shell/issues/3493)

Reply:
My take away is that there is a lot of debate. Switching to an int is probably best in this case. I don't know that it is going to stop me from using size_t in loops to avoid clang warnings though.

Reply:
I agree, I think that it is one case I have seen it mentioned as acceptable. My thought is just not use it in interfaces/function parameters, but if it makes sense to use it internal to functions that is acceptable as you pointed out.